exporter/loadbalancingexporter/loadbalancer.go

```diff
@@ -148,7 +148,11 @@ func (lb *loadBalancer) removeExtraExporters(ctx context.Context, endpoints []st
 	}
 	for existing := range lb.exporters {
 		if !endpointFound(existing, endpointsWithPort) {
-			_ = lb.exporters[existing].Shutdown(ctx)
+			exp := lb.exporters[existing]
+			// Shutdown the exporter asynchronously to avoid blocking the resolver
+			go func() {
+				_ = exp.Shutdown(ctx)
```

Comment:
I'm OK with this, but isn't it cleaner to use `context.WithTimeout` wrapping the context? Would Shutdown respect the timeout?

Reply:
I'm not sure what's the need for this. With all the locks this shutdown can block consumes which supposed to be handled by another available sub-exporter. If we kill it by timeout, we introduce the same panics back. I think we should mark the sub-exporter as "turned off" and shutdown asynchronously once it handles all the current requests.